### PR TITLE
Sync All/Sync None

### PR DESCRIPTION
**Pull request by ticklemepierce**
Purpose
-
Previously if a user wanted to sync all of their projects, or unsync all of their projects they would have to go through and check/uncheck all of their projects.  If they had a lot of projects this may be time consuming and annoying.  Adding Sync All/Sync None buttons will help alleviate these issues.

Changes
-
Added buttons above the tree in the preferences.
screen shot 2015-11-09 at 5 55 53 pm
Each of these buttons is tied to a function that checks/unchecks all of the tree items accordingly.

Side effects
-
This is a very small change that doesn't touch much so there should be no side effects.

[#OSF-5144]

### CODE REVIEW

File: osfoffline/views/rsc/gui/preferences_gui/preferences.ui [hunk 1]
```diff
@@ -253,10 +266,10 @@ p, li { white-space: pre-wrap; }
...
 &lt;html&gt;&lt;head&gt;&lt;meta name=&quot;qrichtext&quot; content=&quot;1&quot; /&gt;&lt;style type=&quot;text/css&quot;&gt;
 p, li { white-space: pre-wrap; }
-&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'Ubuntu'; font-size:11pt; font-weight:400; font-style:normal;&quot;&gt;
-&lt;p style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;This is OSF OFFLINE. Please go ahead and use it and make more software based off of it. Please and Thank You. &lt;/p&gt;
-&lt;p style=&quot;-qt-paragraph-type:empty; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;&lt;br /&gt;&lt;/p&gt;
-&lt;p align=&quot;center&quot; style=&quot; margin-top:0px; margin-bottom:0px; margin-left:0px; margin-right:0px; -qt-block-indent:0; text-indent:0px;&quot;&gt;© Center for Open Science&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
+&lt;/style&gt;&lt;/head&gt;&lt;body style=&quot; font-family:'.Helvetica Neue DeskInterface'; font-size:13pt; font-weight:400; font-style:normal;&quot;&gt;
```
Comment (abought) [inline osfoffline/views/rsc/gui/preferences_gui/preferences.ui]: Was the font change intentional?

(just trying to make sure the entire UI has consistent look-and-feel)

File: osfoffline/views/rsc/preferences_rc.py [hunk 2]
```diff
@@ -107,14 +110,15 @@ def retranslateUi(self, Preferences):
         self.changeFolderButton_2.setText(_translate("Preferences", "Update"))
         self.treeWidget.headerItem().setText(0, _translate("Preferences", "Projects"))
         self.treeWidget.headerItem().setText(1, _translate("Preferences", "Sync"))
+        self.pushButton.setText(_translate("Preferences", "Sync All"))
```
Comment (abought) [inline osfoffline/views/rsc/preferences_rc.py]: Would it be a major hassle to give your items more descriptive names? (similar to how other buttons are named- `changeFolderButton`, `accountLogOutButton`, etc)

That would make it easier to maintain code that connects to these buttons.

**Comment by abought** [comment]
Two questions and some merge conflicts to be resolved. Will pause CR until ready to continue.

(Note that Nan found and fixed a bug while merging; see history for details)

File: osfoffline/views/rsc/gui/preferences_gui/preferences.ui (hunk 1 above)
Comment (ticklemepierce) [inline osfoffline/views/rsc/gui/preferences_gui/preferences.ui]: No, that appears to be OS specific.  I think our current method of doing the GUI (creating a *.ui file and then generating a *.py file from that) may not be completely correct although I am not positive on that.

**Comment by abought** [comment]
Looks close to good (and thanks for fixing the save bug)

In the process of updating the rsc file, one of the comments above was hidden / left unaddressed.

Can you please give the new buttons more descriptive names/IDs in code?

**Comment by abought** [comment]
Ok to merge, @chennan47 .

File: osfoffline/views/rsc/preferences_rc.py [hunk 3]
```diff
@@ -44,32 +44,33 @@ def setupUi(self, Preferences):
...
-        self.groupBox_5.setGeometry(QtCore.QRect(10, 30, 541, 161))
+        self.groupBox_5.setGeometry(QtCore.QRect(10, 20, 521, 211))
         self.groupBox_5.setObjectName("groupBox_5")
         self.changeFolderButton_2 = QtWidgets.QPushButton(self.groupBox_5)
-        self.changeFolderButton_2.setGeometry(QtCore.QRect(430, 50, 99, 31))
+        self.changeFolderButton_2.setGeometry(QtCore.QRect(420, 100, 99, 31))
         self.changeFolderButton_2.setObjectName("changeFolderButton_2")
         self.treeWidget = QtWidgets.QTreeWidget(self.groupBox_5)
-        self.treeWidget.setGeometry(QtCore.QRect(10, 20, 311, 131))
+        self.treeWidget.setGeometry(QtCore.QRect(10, 50, 311, 151))
```
Comment (chennan47) [inline osfoffline/views/rsc/preferences_rc.py]: The setting here doesn't look nice in windows system which actually the two buttons are stick to the top of the syn menu box. I will make some number change when I merge to make it looks better.

Comment (abought) [inline osfoffline/views/rsc/preferences_rc.py]: Nan, can you please include a screenshot?

**Pull request merged**